packages/syft/src/syft/service/service.py

```diff
@@ -50,10 +53,159 @@
+                    autosplat=autosplat,
+                    args=args,
+                    kwargs=kwargs,
+                )
+            result = func(self, *args, **kwargs)
+            if communication_protocol:
+                result, _ = migrate_args_and_kwargs(
+                    args=(result,),
+                    kwargs={},
+                    to_protocol=communication_protocol,
+                )
+                result = result[0]
+            context = kwargs.get("context", None)
+            context = args[0] if context is None else context
+            attrs_to_attach = {
+                "syft_node_location": context.node.id,
+                "syft_client_verify_key": context.credentials,
+            }
+            return attach_attribute_to_syft_object(
+                result=result, attr_dict=attrs_to_attach
+            )
+
+        if autosplat is not None and len(autosplat) > 0:
+            signature = expand_signature(signature=input_signature, autosplat=autosplat)
+
+        config = ServiceConfig(
+            public_path=_path if path is None else path,
+            private_path=_path if private_path is None else private_path,
+            public_name=("public_" + func_name) if name is None else name,
+            method_name=func_name,
+            doc_string=func.__doc__,
+            signature=signature,
+            roles=roles,
+            permissions=["Guest"],
+            warning=warning,
+        )
+        ServiceConfigRegistry.register(config)
+
+        _decorator.__name__ = func.__name__
+        _decorator.__qualname__ = func.__qualname__
+        return _decorator
+
+    return wrapper
 
 class AbstractService:
     node: AbstractNode
     node_uid: UID
+    store: DocumentStore
+    stash: BaseUIDStoreStash
+    
+    def __init__(self, method_params: Optional[Dict[str, Dict[str, Any]]] = {}) -> None:
+        # Add basic CRUD
+        print(f"Creating {self.object_type} service", file=sys.stderr)
+
+        def wrap_service(service_name: str, kwargs: Dict[str, Any]):
+            wrapper = service_method(
+                path=f"{self.object_type}.{service_name}", 
+                private_path=f"{self.object_type}service.{service_name}", 
+                name=service_name, **kwargs)(getattr(self, service_name).__func__)
+            
+            from types import MethodType
+            setattr(self, wrapper.__name__, MethodType(wrapper, self))
+
+        for service_name in ["set", "get", "update", "delete", "restore"]:
+            kwargs = method_params.get(service_name, {})
+            if kwargs.get("skip_method", False):
+                continue
+            wrap_service(service_name=service_name, kwargs=kwargs)
+    
+    @property
+    def object_type(self) -> str:
+        return self.stash.object_type.__canonical_name__.lower()
+    
+    def set(self, context: AuthedServiceContext, obj: Any) -> SyftError | SyftSuccess:
+        # if hasattr(obj, id):
```

Comment:
How are our permissions being checked here. Are they admin only or do they allow any user? Do we check for `owner`, or `write`. Also are we allowing for the `uid` to be set by the user as we do in action objects (where we make sure its unclaimed first).

Many of the `CreateModelType` objects are uid free and generate one during the transform. I dont 100% think this matters too much either way as long as perform the permission checks, but if we do allow users to inject their own UIDs we actually need to make sure they are unique in the entire system (the action store as well) otherwise they could create an object with the same uid as an object in the action store and... do something weird probably.

It could be that the generic `set` is admin only and is only used by developers to write their own `create` object / methods in services although I imagine at least building a consistent pattern on top of `set` for everyone is a good idea, or creating a super().create() might also be helpful.

Reply:
The permissions are checked once when the API is created and once in the store, we could also add some safeguards there as well, though I think for some methods we already have that, but I am not sure how consistent they are. From my understanding, this PR is only for the services and we will refactor `store` as well sometime in the (hopefully near) future. 

Regarding the UID issue, indeed you are right that it might be a problem, we can leave `set` for admin only and add create as well, with the assumption that we have `CreateModelType` object for every object. I could work on adding that if you agree.